### PR TITLE
Updates for removing client commands from server Makefile.

### PR DESCRIPTION
**Pull request by mkraft**
Companion to https://github.com/mattermost/mattermost-server/pull/7892. These changes are minimal pending the new doc site launch.

### CODE REVIEW

File: source/developer/dev-setup-compiling.rst [hunk 1]
```diff
@@ -1,5 +1,22 @@
...
+
+.. code-block:: bash
+
+  # Start the API server (required)
+  cd ~/go/src/github.com/mattermost/mattermost-server
+  make run
+
+  # Start the web app (optional)
+  cd $WEBAPP_DIR
+  make help
```
Comment (crspeller) [inline source/developer/dev-setup-compiling.rst]: You mean `make run`?

File: source/developer/developer-flow.md [hunk 2]
```diff
@@ -10,7 +10,7 @@ Here's a general workflow a Mattermost developer working on our [mattermost-serv
 1. Take a look at the [Repository structure](https://docs.mattermost.com/developer/developer-flow.html#repository-structure) to find out where to look for what you're working on.
 2. On your fork, create a branch `PLT-####` where #### is the ticket number if it is a [Jira](https://mattermost.atlassian.net) ticket, or `GH-####` if it is a GitHub Issue without a Jira ticket.
 3. Make the code changes required to complete your ticket, making sure to write or modify unit tests where appropriate.
-4. To test your changes, run `make run` from the root directory of the server respository. This will start up the server and a watcher process that will build any changes to the client as you make them. To get changes to the server it must be restarted with `make restart-server`. Your server will be running at `http://localhost:8065`.
+4. To run your code, run `make run` from the root directory of the server respository. This will start up the server and a watcher process that will build any changes to the client as you make them. To get changes to the webapp server it must be restarted with `make restart`. Your webapp server will be running at `http://localhost:8065`.
```
Comment (crspeller) [inline source/developer/developer-flow.md]: But it won't do this. You have to cd into the webapp directory and run it there too.

File: source/developer/dev-setup-compiling.rst [hunk 3]
```diff
@@ -1,5 +1,22 @@
...
+
+  # Make docs from either repo
+  make help
+  
+
 Compiling Mattermost
```
Comment (crspeller) [inline source/developer/dev-setup-compiling.rst]: This section should be modified to mention that you need to do `make run` in both now.

File: source/developer/developer-flow.md [hunk 4]
```diff
@@ -10,7 +10,7 @@ Here's a general workflow a Mattermost developer working on our [mattermost-serv
...
 5. Once everything works to meet the ticket requirements, stop mattermost running `make stop` in the server repository and then run `make check-style` for check your syntax and `make test` for run the test in both repositories, server and webapp.
```
Comment (crspeller) [inline source/developer/developer-flow.md]: This needs modification. `make check-style` and `make test` need to be run in both repositories if you have made changes to both. Or just the specific one you modified.

File: source/developer/developer-flow.md [hunk 5]
```diff
@@ -20,12 +20,17 @@ Here's a general workflow a Mattermost developer working on our [mattermost-serv
...
+* `make help` list all of the make commands
+* `make run` will run only the server and not the client
 * `make clean-docker` stops and removes your Docker images and is a good way to wipe your database
-* `make run-fullmap` will run the server and build the client with the full source map for easier debugging
 * `make clean` cleans your local environment of temporary files
 * `make nuke` wipes your local environment back to a completely fresh start
 
+### Useful webapp Makefile commands ###
+
+* `make help` list all of the make commands
```
Comment (crspeller) [inline source/developer/developer-flow.md]: `make run`?

File: source/developer/running-mattermost.rst [hunk 6]
```diff
@@ -1,6 +1,11 @@
 Running Mattermost
```
Comment (crspeller) [inline source/developer/running-mattermost.rst]: This file needs modification. For instance "To check styles of Go and JavaScript" needs to be separate for both repositories.

File: source/developer/developer-flow.md (hunk 5 above)
Comment (mkraft) [reply source/developer/developer-flow.md]: This one is meant to be `help`.

File: source/developer/dev-setup-compiling.rst (hunk 3 above)
Comment (mkraft) [reply source/developer/dev-setup-compiling.rst]: What's a sentence in english in your words that would explain what we want a user to know how to do at the end of this section?

File: source/developer/running-mattermost.rst (hunk 6 above)
Comment (mkraft) [reply source/developer/running-mattermost.rst]: Same question for "Running Mattermost". Is this a document to get a user up-and-running to run the webapp locally with the server? There is a section `To run changes to client-side code:` so I'm just wondering if you have specific thoughts on what's lacking or what's present that should be there?

File: source/developer/developer-flow.md (hunk 2 above)
Comment (jwilander) [reply source/developer/developer-flow.md]: Agreed, also it's not running a "webapp server". The server is serving the webapp but it's doing a lot more than just that

**Comment by mkraft** [comment]
Closing this then until we're ready to do the work once properly on the new docs site.

**Pull request closed without merging**